apphub/contrastive_learning/simclr/simclr.ipynb

```diff
@@ -0,0 +1,599 @@
+      "FastEstimator-Train: step: 4500; NTXent: 11.964204; steps/sec: 4.92;\n",
+      "FastEstimator-ModelSaver: Saved model to /tmp/tmp33wb3rot/model_epoch_46.h5\n",
+      "FastEstimator-Train: step: 4508; epoch: 46; contrastive_accuracy: 0.9936; epoch_time: 19.92 sec;\n",
+      "FastEstimator-Train: step: 4600; NTXent: 11.970972; steps/sec: 4.93;\n",
+      "FastEstimator-ModelSaver: Saved model to /tmp/tmp33wb3rot/model_epoch_47.h5\n",
+      "FastEstimator-Train: step: 4606; epoch: 47; contrastive_accuracy: 0.99444; epoch_time: 19.9 sec;\n",
+      "FastEstimator-Train: step: 4700; NTXent: 11.972122; steps/sec: 4.92;\n",
+      "FastEstimator-ModelSaver: Saved model to /tmp/tmp33wb3rot/model_epoch_48.h5\n",
+      "FastEstimator-Train: step: 4704; epoch: 48; contrastive_accuracy: 0.9936; epoch_time: 19.93 sec;\n",
+      "FastEstimator-Train: step: 4800; NTXent: 11.961956; steps/sec: 4.93;\n",
+      "FastEstimator-ModelSaver: Saved model to /tmp/tmp33wb3rot/model_epoch_49.h5\n",
+      "FastEstimator-Train: step: 4802; epoch: 49; contrastive_accuracy: 0.99414; epoch_time: 19.89 sec;\n",
+      "FastEstimator-Train: step: 4900; NTXent: 11.124333; steps/sec: 4.91;\n",
+      "FastEstimator-ModelSaver: Saved model to /tmp/tmp33wb3rot/model_epoch_50.h5\n",
+      "FastEstimator-Train: step: 4900; epoch: 50; contrastive_accuracy: 0.99446; epoch_time: 19.94 sec;\n",
+      "FastEstimator-Finish: step: 4900; model_lr: 0.001; total_time: 1055.67 sec;\n"
+     ]
+    }
+   ],
+   "source": [
+    "traces = [\n",
+    "    Accuracy(true_key=\"label\", pred_key=\"logit\", mode=\"train\", output_name=\"contrastive_accuracy\"),\n",
+    "    ModelSaver(model=model_con, save_dir=save_dir)\n",
+    "]\n",
+    "\n",
+    "estimator_pretrain = fe.Estimator(pipeline=pipeline_pretrain,\n",
+    "                                  network=network_pretrain,\n",
+    "                                  epochs=epochs_pretrain,\n",
+    "                                  traces=traces,\n",
+    "                                  max_train_steps_per_epoch=max_train_steps_per_epoch,\n",
+    "                                  monitor_names=\"contrastive_accuracy\")\n",
+    "estimator_pretrain.fit()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "curious-circus",
+   "metadata": {},
+   "source": [
+    "## Finetune the model on image classification task\n",
+    "\n",
+    "Once the model is pretrained, we can finetune the model on specific task. In this case we are going to use this pretrained model on CIFAR10 image classification. Remember in the previous section we build the `model_con` and `model_finetune`. Because those two models share the same encoder object, by training the `model_con` (we just did), the encoder of `model_fintune` is also trained. The finetuing of the model is literally just supervised training with the pretrained encoder. In order to demonstrate the benefit of SimCLR, we are going to trained the\n",
```

Comment:
```suggestion
    "Once the model is pretrained, we can finetune the model on a specific task. In this case we are going to use this pretrained model on ciFAIR10 image classification. Remember in the previous section we built both `model_con` and `model_finetune`. Because those two models share the same encoder object, by (pre)training the `model_con`, the encoder of `model_fintune` is also trained. The finetuing of the model is literally just supervised training with the pretrained encoder. In order to demonstrate the benefit of SimCLR, we are going to train the network with only 10% of the labeled training data and compare with how well the model could do without SimCLR training.\n",
```